--- gcc/expr.c
+++ gcc/expr.c
@@ -126,1 +126,1 @@
-static void move_by_pieces_1 (rtx (*) (rtx, ...), enum machine_mode,
+static void move_by_pieces_1 (rtx (*) (), enum machine_mode,
@@ -135,1 +135,1 @@
-static void store_by_pieces_2 (rtx (*) (rtx, ...), enum machine_mode,
+static void store_by_pieces_2 (rtx (*) (), enum machine_mode,
@@ -1050,1 +1050,1 @@
-move_by_pieces_1 (rtx (*genfun) (rtx, ...), enum machine_mode mode,
+move_by_pieces_1 (rtx (*genfun) (), enum machine_mode mode,
@@ -2611,1 +2611,1 @@
-store_by_pieces_2 (rtx (*genfun) (rtx, ...), enum machine_mode mode,
+store_by_pieces_2 (rtx (*genfun) (), enum machine_mode mode,
